### DISCUSSION TITLE
creating visual stimulus with bon vision

### DISCUSSION THREAD

**Discussion by sagi-idan** [bonsai-rx/bonsai#1403] [Q&A]
Hi there.
I am trying to create a visual stimulus in 8 directions + blanc, I used the the attached file and encounters two problems.
1. the angle get scrambled and not as they should appear after the permutation node and not on 45 degrees increment
2. the Log event node doesn't  log the angle

Thanks in advance.
  
[pref_ang_aqc_4.zip](https://github.com/bonsai-rx/bonsai/files/11665598/pref_ang_aqc_4.zip)

**Comment by bruno-f-cruz** [comment]
1. You are inputting `Angle` in degrees instead of Radians. Instead of `Multiply(45)` use `Multiply (0.78539816339)` and check if it gives you what you expect.
2. Seems to work on my PC...
```
Frame,Timestamp,Value
340,5.6837716999999994,0
699,11.6751209,360
1060,17.692795,270
```

**Reply by sagi-idan** [reply]
Hi.
Thanks for your response but it still doesn't work well.
The objective of my workflow is to create a series of 8 gratings direction + blanc trial without contrast. I want to have 2sec of dark time, 2 sec of stimulus and 2 sec of dark time again in each trail.

I tried to use the attached **pref_ang_acq_2** but instead of getting 9 trail of (8 different gratings directions + 1 blank) 10 trails appear and the angles aren't getting logged correctly, I attached the csv log file with extra column of the actual sequence that appeared on the screen.

I also trued to use
  pref_ang_acq_5 but no matter where I put the LogEvent node in the workflow it didn't record the angle.

I will be most thankful for your help.   
[workflows.zip](https://github.com/bonsai-rx/bonsai/files/11689865/workflows.zip)
[Events.csv](https://github.com/bonsai-rx/bonsai/files/11689867/Events.csv)
